File: 19 Favorite Code Examples/BatchProcess/SolidWorksMacro.vb
Imports SolidWorks.Interop.sldworks
Imports SolidWorks.Interop.swconst
Imports System
Imports System.Windows.Forms

Partial Class SolidWorksMacro

    Public Sub main()
        Dim MyForm As New Form1
        Dim DiaRes As DialogResult = MyForm.ShowDialog
        If DiaRes = DialogResult.OK Then
            'process the folders
            Module1.Process(MyForm.FromFolderTextBox.Text, _
              MyForm.OutputTextBox.Text)
        End If
    End Sub

    ''' <summary>
    ''' The SldWorks swApp variable is pre-assigned for you.
    ''' </summary>
    Public swApp As SldWorks


End Class

File: 19 Favorite Code Examples/BatchProcess/SolidWorksMacro.cs
using System.Windows.Forms;
using SolidWorks.Interop.sldworks;

namespace BatchProcess
{
    public partial class SolidWorksMacro
    {
        public void Main()
        {
            Form1 frm = new Form1();
            if (frm.ShowDialog() == DialogResult.OK)
            {
                string filePath = frm.textBoxSource.Text;
                string outputPath = frm.textBoxOutput.Text;
                Batch.swApp = swApp;  //pass the SOLIDWORKS application to the static class
                Batch.Process(filePath, outputPath);
            }

            return;
        }

        // The SldWorks swApp variable is pre-assigned for you.
        public SldWorks swApp;

    }
}

